### PR TITLE
Start-tasks: wait on xtriggers

### PR DESCRIPTION
**Pull request by hjoliver**
Start-tasks (`cylc play --start-task=ID`) are important for workflow DR failover.

Currently they get force-triggered at start-up, to run immediately. They should be spawned via setting all prerequisites instead, to wait on clock- and xtriggers. (c.f. clock-triggered tasks at the beginning of the graph, for normal start up).

(This came up in NIWA ops Cylc 8 migration, and @dwsutherland has had to work around it).

Very small change (+3/-1 lines code change + tests) but fairly important so should go in 8.4 IMO.

**Check List**

- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Applied any dependency changes to both `setup.cfg` (and `conda-environment.yml` if present).
- [x] Tests are included (or explain why tests are not needed).
- [x] Changelog entry included if this is a change that can affect users
- [x] [Cylc-Doc](https://github.com/cylc/cylc-doc) pull request opened if required at cylc/cylc-doc/pull/XXXX.
- [x] If this is a bug fix, PR should be raised against the relevant `?.?.x` branch.

### CODE REVIEW

**Comment by hjoliver** [comment]
Example:
```cylc
[scheduler]
    cycle point format = %Y
[scheduling]
    initial cycle point = 2024  # current year
    [[graph]]
        P1Y = "@wall_clock => a"
[runtime]
    [[a]]
```
Normal start:
- `cylc vip` - 2024/a runs, 2025/a waits on the clock ✔️ 

Start-task, master:
- `cylc vip --start=task=2025/a` - 2025/a runs immediately despite the unsatisfied clock trigger  ❌ 

Start-task, this branch
- `cylc vip --start=task=2025/a` - 2025/a spawns but waits on the clock ✔️

**Comment by hjoliver** [comment]
(One functional test failing due to the change in behaviour: `tests/f/cylc-play/05-start-task.t`)

**Review by dwsutherland — APPROVED** [review]
LGTM.. Much needed thanks 👍

File: tests/integration/test_task_pool.py [hunk 1]
```diff
@@ -2305,3 +2305,80 @@ async def test_job_insert_on_crash(one_conf, flow, scheduler, start):
...
+        assert (
+            set(itask.identity for itask in itasks) == set([
+                "2050/foo",
+                "2051/foo",
+                "2050/bar",
+                "2050/baz",
+            ])
+        )
+
+        # Check xtriggers and release tasks if ready ot run
```
Comment (hjoliver) [inline tests/integration/test_task_pool.py:2368]:
```suggestion
        # Check xtriggers
```

**Review by wxtim — APPROVED** [review]

**Pull request merged**